text_extensions_for_pandas/array/arrow_conversion.py

```diff
@@ -154,13 +158,21 @@ def arrow_to_span(extension_array: pa.ExtensionArray) -> SpanArray:
             raise ValueError("Only pyarrow.Array with a single chunk is supported")
         extension_array = extension_array.chunk(0)
 
+    # NOTE: workaround for bug in parquet reading
+    if pa.types.is_struct(extension_array.type):
+        index_dtype = extension_array.field(ArrowSpanType.BEGINS_NAME).type
+        target_text_dict_dtype = extension_array.field(ArrowSpanType.TARGET_TEXT_DICT_NAME).type
+        extension_array = pa.ExtensionArray.from_storage(
+            ArrowSpanType(index_dtype, target_text_dict_dtype),
+            extension_array)
+
     assert pa.types.is_struct(extension_array.storage.type)
 
-    # Get target text from the begins field metadata and decode string
-    metadata = extension_array.storage.type[ArrowSpanType.BEGINS_NAME].metadata
-    target_text = metadata[ArrowSpanType.TARGET_TEXT_KEY]
-    if isinstance(target_text, bytes):
-        target_text = target_text.decode()
+    # Create target text StringTable and text_ids from dictionary array
+    target_text_dict_array = extension_array.storage.field(ArrowSpanType.TARGET_TEXT_DICT_NAME)
+    target_texts = [s.as_py() for s in target_text_dict_array.dictionary]
+    string_table, _ = StringTable.merge_things(target_texts)
```

Comment:
This assumes that the `text_ids` will match the new ids returned from `merge_thing`. I think it's safe to do this.

Reply:
yup, that's a good idea. I'll add that here.

Reply:
It's a little more useful to have a method to create a table from just a list of strings, so I added that. If you think we should have just one that takes things and ids to make a table, I can change it.